mantidproject/mantid
# New LeanElasticPeak class

**Description of work.**

This is the beginning of the work to make working with Peaks easier, in particular for reactor instruments. This creates a new peak type, a LeanElasticPeak, which doesn't have any instrument information, everything is base on Q-sample.

A BasePeak was created which includes all the common parts of LeanElasticPeak and Peak, both of which inherit from BasePeak.

**Motivation**

Once your data is converted into Q-sample there are very few reasons you need to know anything about the instrument at that point, you should be able to find peaks, find UB, predict peaks and integrate peaks without going back to the instrument.

**To test:**

Since this isn't adding anything actually usable at the moment there isn't much to directly test. This is mainly code review, look at the tests and see if they make sense. 


Once the LeanElasticPeak is usable, probably the next PR with the LeanElasticPeaksWorkspace, all the documentation will be updated.




---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
